homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,381 @@
+                )
+
+        # Connection sensors
+        connection_datas: Dict[str, any] = await self._api.connection.get_status()
+        for sensor_key, sensor_attrs in CONNECTION_SENSORS.items():
+            if self._connection_sensors.get(sensor_key) is not None:
+                # Seen sensor -> updating
+                _LOGGER.debug("Updating Freebox sensor: %s", sensor_key)
+                self._connection_sensors[sensor_key].update(
+                    connection_datas[sensor_key]
+                )
+            else:
+                # New sensor, should be unique
+                _LOGGER.debug("Adding Freebox sensor: %s", sensor_key)
+                self._connection_sensors[sensor_key] = FreeboxSensor(sensor_attrs)
+                self._connection_sensors[sensor_key].update(
+                    connection_datas[sensor_key]
+                )
+
+        self._attrs = {
+            "IPv4": connection_datas.get("ipv4"),
+            "IPv6": connection_datas.get("ipv6"),
+            "connection_type": connection_datas["media"],
+            "uptime": datetime.fromtimestamp(
+                round(datetime.now().timestamp()) - syst_datas["uptime_val"]
+            ),
+            "firmware_version": self.firmware_version,
+            "serial": syst_datas["serial"],
+        }
+
+        dispatcher_send(self.hass, SENSOR_UPDATE)
+
+    async def reboot(self) -> None:
+        """Reboot the Freebox."""
+        await self._api.system.reboot()
+
+    async def close(self) -> None:
+        """Close the connection."""
+        if self._api is not None:
+            await self._api.close()
+        self._api = None
+
+    @property
+    def name(self) -> str:
+        """Return the router name."""
+        return self._name
+
+    @property
+    def manufacturer(self) -> str:
+        """Return the router manufacturer."""
+        return "Freebox SAS"
+
+    @property
+    def mac(self) -> str:
+        """Return the router MAC address."""
+        return self._mac
+
+    @property
+    def firmware_version(self) -> str:
+        """Return the router software version."""
+        return self._sw_v
+
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self.mac)},
+            "identifiers": {(DOMAIN, self.mac)},
+            "name": self.name,
+            "manufacturer": self.manufacturer,
+            "sw_version": self.firmware_version,
+        }
+
+    @property
+    def devices(self) -> Dict[str, any]:
+        """Return all devices."""
+        return self._devices
+
+    @property
+    def sensors(self) -> Dict[str, any]:
+        """Return all sensors."""
+        return {**self._temperature_sensors, **self._connection_sensors}
+
+    @property
+    def wifi(self) -> Wifi:
+        """Return the wifi."""
+        return self._api.wifi
+
+
+class FreeboxDevice:
```

Comment:
This class doesn't look necessary. Just merge this with the tracker entity class.

Reply:
Will merge `FreeboxDevice` and `FreeboxTrackerEntity`.

I thought it wasn't allowed.